fixing internal test failure on non sm_80 machines

HDCharles opened the pull request:
Summary:
These tests were failing on non sm_80+ machines used for internal CI, added check to skip this.

D48295360 added new tests that work in OSS but not in phabricator CI

https://www.internalfb.com/intern/test/562950057441807?ref_report_id=0

https://www.internalfb.com/intern/test/281475080709193?ref_report_id=0

Test Plan: see phabricator result

Differential Revision: D48417499



cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

peterbell10 commented on test/inductor/test_pattern_matcher.py:
```diff
@@ -63,6 +63,9 @@ def _test_mixed_impl(self, fn, args, mixed_mm_expected, fallback_mixed_mm_expect
         self.assertEqual("mixed_mm" in code, mixed_mm_expected)
         self.assertEqual("fallback_mixed_mm" in code, fallback_mixed_mm_expected)
 
+    @unittest.skipIf(
+        any(int(arch[3:]) < 80 for arch in torch.cuda.get_arch_list()), "need sm_80"
+    )
```
Looks like you want
```python
from torch.testing._internal.common_cuda import SM80OrLater
```

HDCharles replied:
oh thanks, yeah i didn't know that existed

HDCharles committed:
fixing internal test failure on non sm_80 machines (#107340)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/107340

These tests were failing on non sm_80+ machines used for internal CI, added check to skip this.

D48295360 added new tests that work in OSS but not in phabricator CI

https://www.internalfb.com/intern/test/562950057441807?ref_report_id=0

https://www.internalfb.com/intern/test/281475080709193?ref_report_id=0

Test Plan: see phabricator result

Reviewed By: davidberard98

Differential Revision: D48417499

fbshipit-source-id: 47c9b531f11e59a9290989b7c5e85c1b4beed71e

HDCharles commented:
@pytorchmergebot merge
